app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,72 @@
+class TweetController < ApplicationController
+
+  get '/tweets' do
+    if logged_in?
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'tweets/new_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if missing_field?
+      erb :'tweets/new_tweet', locals: {missing_field: "Please enter a tweet."}
+    else
+      @user = current_user
+      @user.tweets.create(params[:data])
+      redirect "/tweets"
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect 'login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+     @tweet = Tweet.find(params[:id])
+     erb :'tweets/edit_tweet'
+    else
+      redirect 'login'
+    end
+  end
+
+  post '/tweets/:id' do
+    if content_unchanged?
+      @tweet = Tweet.find(params[:id])
+      erb :'tweets/edit_tweet', locals: {content_unchanged: "Please change the content of your tweet."}
+    else
+      if logged_in?
+        @tweet = Tweet.find(params[:id])
+        @tweet.update(params[:data])
+        redirect "/tweets"
+      end
+    end
+  end
+
+  post '/tweets/:id/delete' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      if @tweet.user_id == session[:user_id]
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(:id => params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the users tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully simply returning nil if the user is trying to delete a tweet that they do not have access to.